marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/provenance/prov-lib.sjs

```diff
@@ -48,6 +49,131 @@ function deleteProvenance(deleteRequest, endpointState) {
+    '              <object>\n' +
+    '                <val>$obj</val>\n' +
+    '              </object>\n' +
+    '            </triple>\n' +
+    '          </triples>\n' +
+    '        </template>\n' +
+    '        <template>\n' +
+    '          <context>*[fn:namespace-uri(.) != "http://www.w3.org/ns/prov#"]</context>\n' +
+    '          <triples>\n' +
+    '            <triple>\n' +
+    '              <subject>\n' +
+    '                <val>$id</val>\n' +
+    '              </subject>\n' +
+    '              <predicate>\n' +
+    '                <val>sem:iri(fn:namespace-uri(.)||fn:local-name(.))</val>\n' +
+    '              </predicate>\n' +
+    '              <object>\n' +
+    '                <val>xs:string(string(.))</val>\n' +
+    '              </object>\n' +
+    '            </triple>\n' +
+    '          </triples>\n' +
+    '        </template>\n' +
+    '      </templates>\n' +
+    '    </template>\n' +
+    '    <template>\n' +
+    '      <context>prov:used</context>\n' +
+    '      <triples>\n' +
+    '        <triple>\n' +
+    '          <subject>\n' +
+    '            <val>sem:iri(prov:activity/@prov:ref)</val>\n' +
+    '          </subject>\n' +
+    '          <predicate>\n' +
+    '            <val>sem:iri("http://www.w3.org/ns/prov#used")</val>\n' +
+    '          </predicate>\n' +
+    '          <object>\n' +
+    '            <val>sem:iri(prov:entity/@prov:ref)</val>\n' +
+    '          </object>\n' +
+    '        </triple>\n' +
+    '      </triples>\n' +
+    '    </template>\n' +
+    '    <template>\n' +
+    '      <context>prov:*[@prov:id][prov:label]</context>\n' +
+    '      <vars>\n' +
+    '        <var>\n' +
+    '          <name>id</name>\n' +
+    '          <val>sem:iri(@prov:id)</val>\n' +
+    '        </var>\n' +
+    '      </vars>\n' +
+    '      <templates>\n' +
+    '        <template>\n' +
+    '          <context>prov:label</context>\n' +
+    '          <vars>\n' +
+    '            <var>\n' +
+    '              <name>obj</name>\n' +
+    '              <val>string(.)</val>\n' +
+    '            </var>\n' +
+    '          </vars>\n' +
+    '          <triples>\n' +
+    '            <triple>\n' +
+    '              <subject>\n' +
+    '                <val>$id</val>\n' +
+    '              </subject>\n' +
+    '              <predicate>\n' +
+    '                <val>sem:iri("http://www.w3.org/ns/prov#label")</val>\n' +
+    '              </predicate>\n' +
+    '              <object>\n' +
+    '                <val>$obj</val>\n' +
+    '              </object>\n' +
+    '            </triple>\n' +
+    '          </triples>\n' +
+    '        </template>\n' +
+    '      </templates>\n' +
+    '    </template>\n' +
+    '  </templates>\n' +
+    '</template>');
+  const templateUri = '/hub-template/RecordProvenance.xml';
+  const permissions = [
+    xdmp.permission('ps-user', 'read'),
+    xdmp.permission('ps-internal', 'update')
```

Comment:
Sorry, @rahulvudutala. I was thinking of the provenance documents themselves when I recommend the update permission. Giving "data-hub-developer" update permissions would be best for the TDE.

Reply:
Everything else looks good.